### ISSUE TITLE
The behavior of `ModulatedDeformConv` when the mask shape is different from the input shape.

### ISSUE THREAD

**Issue by KeitaOtani** [microsoft/GLIP#52]
Thank you for your great work.

When I changed `maskrcnn_benchmark.layers.ModulatedDeformConv` to `torch.ops.DeformConv2d` to avoid CUDA compilation, I found a strange behavior.

In `maskrcnn_benchmark/modeling/rpn/vldyhead.py` at line 210, the input shape is different from the shape of offset and mask.
The input shape is the shape of `level + 1`, but the offset and mask shapes are the shape of `level`.
But it does not cause an error.

I guess it is because `ModulatedDeformConv` does not check the shapes of the mask and offset.
I do not read its CUDA code, but I found strange behavior in it.
```
import torch
from maskrcnn_benchmark.layers import ModulatedDeformConv

moddc = ModulatedDeformConv(8, 16, 3).cuda()
x = torch.randn(1, 8, 10, 10).cuda()
offset = torch.randn(1, 18, 20, 20).cuda()
mask = torch.randn(1, 9, 20, 20).cuda()

res1 = moddc(x, offset, mask)
res2 = moddc(x, offset.reshape(4, 18, 10, 10)[0], mask.reshape(4, 9, 10, 10)[0])
result = torch.abs(res1 - res2).max()
```
The result is always 0.
`vldyhead.py` has the same behavior that may not be desirable.

I am looking forward to your reply.
Sincerely,

**Comment by Haotian-Zhang** [comment]
Since the deformable convolution module is not designed by ours, please refer to the original maskrcnn_benchmark or PyTorch repo for the details. Thanks!

**Comment by KeitaOtani** [comment]
Thank you for your reply.
I understood that the deformable convolution module does not check the input shape, and you didn't create the module.
But, at `maskrcnn_benchmark/modeling/rpn/vldyhead.py#L210`, the input `offset` and `mask` shapes are different from the output shape of the deform convolution.
I think that not throwing an error here is responsible for the deformable convolution module, but inputting the wrong shape is not.
If I make a mistake, please let me know.
Sincerely,

**Comment by Li-fAngyU** [comment]
Same issue here!

**Comment by KeitaOtani** [comment]
Thanks for your comment, @Li-fAngyU!
I will submit this issue again.

**Comment by Li-fAngyU** [comment]
Thanks for your reply, @KeitaOtani!
Here is the code (Hope this helps):
```python
import torch
from maskrcnn_benchmark.layers import ModulatedDeformConv

input = torch.rand(1, 256, 14, 14)
kh, kw = 3, 3
offset = torch.rand(1, 2 * kh * kw, 28, 28)
mask = torch.rand(1, kh * kw, 28, 28)

glip_dcn= ModulatedDeformConv(
    in_channels=256,
    stride = 1,
    out_channels=256,
    kernel_size=[kh, kw],
    padding=1,
    groups=1

import torchvision
tochvision_dcn= torchvision.ops.DeformConv2d(
    in_channels=256,
    stride = 1,
    out_channels=256,
    kernel_size=[kh, kw],
    padding=1,
    )

glip_dcn= glip_dcn.to('cuda')
out_glip= glip_dcn(input, offset, mask)
print('------------------')
print('out_glip.shape:',out.shape, 'input_glip shape:',input.shape, 'offset_glip shape:', offset.shape, 'mask_glip shape:', mask.shape)
print(glip_dcn)
print('------------------')

try:
    tochvision_dcn= tochvision_dcn.to('cuda')
    out_torchvision = tochvision_dcn(input, offset, mask)
    print('------------------')
    print('out_torchvision.shape:',out_torchvision.shape, 'input_torchvision shape:',input.shape, 'offset_torchvision shape:', offset.shape, 'mask_torchvision shape:', mask.shape)
    print('------------------')
except:
    print('torchvision.ops.DeformConv2d error')
    print('torchvision.ops.DeformConv2d: ',deform_conv_torch)
```
And the output result:
```
------------------
out_glip.shape: torch.Size([1, 256, 14, 14]) input_glip shape: torch.Size([1, 256, 14, 14]) offset_glip shape: torch.Size([1, 18, 28, 28]) mask_glip shape: torch.Size([1, 9, 28, 28])
ModulatedDeformConv(in_channels=256, out_channels=256, kernel_size=(3, 3), stride=1, dilation=1, padding=1, groups=1, deformable_groups=1, bias=True)
------------------
torchvision.ops.DeformConv2d error
torchvision.ops.DeformConv2d:  DeformConv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
```

BTW, Did you mention this issue on this original maskrcnn_benchmark or PyTorch repo?

**Comment by KeitaOtani** [comment]
Thanks for your code, @Li-fAngyU!
I will not submit this issue to the maskrcnn_benchmark repository.
I only focus on the problem that the shape of the input for `ModulatedDeformConv` is wrong, and therefore the back part of the input mask is not used vldyhead.py.
